### PR TITLE
[azservicebus] Making a separate version of amqp-common as the `internal` package for messaging

### LINKED ISSUE #15646: [amqp-common] Migrate Azure/azure-amqp-common-go into monorepo
Service Bus (track 0 and track 2) as well as the GA'd Event Hubs package both depend on this external dependency.

This is purely an internally used package (unlike `go-amqp`) so moving this into the monorepo (azure-sdk-for-go) makes sense.

For SB, which is in the monorepo, this isn't a problem. However, we are probably stymied from doing this immediately because of Event Hubs, which is in an external repo and wouldn't be able to use internal packages in azure-sdk-for-go (yet).

### PR DESCRIPTION
**Pull request by richardpark-msft**
The messaging SDKs like Service Bus (and eventually EventHubs) share common code for authenticating using cbs/claims, and various things related to AMQP. This was all bundled up into a package located at github.com/Azure/azure-amqp-common-go. 

We don't actually need to do that anymore for SB, and it's desirable to move more code into the monorepo.

This is a pretty minimal copy - just what's latest plus some changes to make it build as an `internal` package.

Fixes #15646

### CODE REVIEW

File: sdk/messaging/internal/version.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package internal
+
+const (
+	// Version is the semantic version of the library
+	Version = "3.1.1"
```
Comment (jhendrixMSFT) [inline sdk/messaging/internal/version.go:7]: This should be `0.1.0` yes?

File: sdk/messaging/internal/uuid/uuid.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package uuid
```
Comment (jhendrixMSFT) [inline sdk/messaging/internal/uuid/uuid.go:1]: We already have a `uuid` package in the core `internal` module.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Also need a LICENSE.txt file and doc.go

File: sdk/messaging/internal/uuid/uuid.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/messaging/internal/uuid/uuid.go]: Done.

File: sdk/messaging/internal/version.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/internal/version.go]: Yes

**Comment by richardpark-msft** [comment]
> 
> 
> Also need a LICENSE.txt file and doc.go

Done :)

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**